### PR TITLE
[GraphOptimizer] Do transpose and reshape optimizations first

### PR DESCRIPTION
**Pull request by aturetsk**
*Description*:
This is a cleanup rather than a functional change.
Transpose and Reshape nodes may prevent other optimizations from triggering (e.g. BatchNorm + Convolution merge, Transpose + MatMul merge). As optimizeTranspose and optimizeReshape do not depend on other optimizations, it makes sense to do them first, so that Transposes and Reshapes would be optimized out and let other optimizations work.
I did some optimization re-ordering related to reshape/transpose in my previous PRs (#1680, #1782). This is just a cleaner way to handle it. I think it's better to put the optimizations in the beginning and have an explicit comment explaining why we are doing this, rather than have implicit optimization ordering.

*Testing*:
No new tests, since it's just a clean up.

*Documentation*:
N/A

### CODE REVIEW

**Comment by nadavrot** [comment]
LGTM. Can you please mention the name of the networks that become faster? For example, "with this change TensorFlow's AlexNet is now optimized".

File: lib/Optimizer/GraphOptimizer.cpp [hunk 1]
```diff
@@ -2092,6 +2092,13 @@ void glow::optimize(Function *F, CompilationMode mode) {
     DCE(F);
   }
 
+  // Reshapes and transposes can prevent other optimizations from triggering, so
+  // try to optimize them out first.
+  optimizeReshape(F);
+  if (mode == CompilationMode::Infer) {
+    optimizeTranspose(F);
```
Comment (rdzhabarov) [inline lib/Optimizer/GraphOptimizer.cpp]: what is the reason transposes are not eliminated in training?

**Comment by aturetsk** [comment]
@nadavrot, this is just a cleanup, so there is no performance improvement. I just tried to make it more obvious why some optimizations are done after other optimizations, by re-organizing code a bit.

@rdzhabarov, renamed optimizeTranspose to transposeConstants.
